sparse_strips/vello_hybrid/src/render.rs

```diff
@@ -0,0 +1,268 @@
+        let fill_rule = Fill::NonZero;
+        let paint = BLACK.into();
+        let stroke = Stroke {
+            width: 1.0,
+            join: Join::Bevel,
+            start_cap: Cap::Butt,
+            end_cap: Cap::Butt,
+            ..Default::default()
+        };
+        let blend_mode = BlendMode::new(Mix::Normal, Compose::SrcOver);
+
+        Self {
+            width: width.into(),
+            height: height.into(),
+            wide,
+            alphas,
+            line_buf,
+            tiles,
+            strip_buf,
+            transform,
+            paint,
+            fill_rule,
+            stroke,
+            blend_mode,
+        }
+    }
+
+    /// Fill a path with the current paint and fill rule.
+    pub fn fill_path(&mut self, path: &BezPath) {
+        flatten::fill(path, self.transform, &mut self.line_buf);
+        self.render_path(self.fill_rule, self.paint.clone());
+    }
+
+    /// Stroke a path with the current paint and stroke settings.
+    pub fn stroke_path(&mut self, path: &BezPath) {
+        flatten::stroke(path, &self.stroke, self.transform, &mut self.line_buf);
+        self.render_path(Fill::NonZero, self.paint.clone());
+    }
+
+    /// Fill a rectangle with the current paint and fill rule.
+    pub fn fill_rect(&mut self, rect: &Rect) {
+        self.fill_path(&rect.to_path(DEFAULT_TOLERANCE));
+    }
+
+    /// Stroke a rectangle with the current paint and stroke settings.
+    pub fn stroke_rect(&mut self, rect: &Rect) {
+        self.stroke_path(&rect.to_path(DEFAULT_TOLERANCE));
+    }
+
+    /// Set the blend mode for subsequent rendering operations.
+    pub fn set_blend_mode(&mut self, blend_mode: BlendMode) {
+        self.blend_mode = blend_mode;
+    }
+
+    /// Set the stroke settings for subsequent stroke operations.
+    pub fn set_stroke(&mut self, stroke: Stroke) {
+        self.stroke = stroke;
+    }
+
+    /// Set the paint for subsequent rendering operations.
+    pub fn set_paint(&mut self, paint: Paint) {
+        self.paint = paint;
+    }
+
+    /// Set the fill rule for subsequent fill operations.
+    pub fn set_fill_rule(&mut self, fill_rule: Fill) {
+        self.fill_rule = fill_rule;
+    }
+
+    /// Set the transform for subsequent rendering operations.
+    pub fn set_transform(&mut self, transform: Affine) {
+        self.transform = transform;
+    }
+
+    /// Reset the transform to identity.
+    pub fn reset_transform(&mut self) {
+        self.transform = Affine::IDENTITY;
+    }
+
+    /// Reset all rendering state to default values.
+    pub fn reset(&mut self) {
+        self.wide.reset();
+    }
+
+    /// Render the current content to a pixmap.
+    pub async fn render_to_pixmap(&self, pixmap: &mut Pixmap) {
+        let render_data = self.prepare_render_data();
+        let mut renderer = Renderer::new(RenderTarget::Headless {
+            width: pixmap.width as u32,
+            height: pixmap.height as u32,
+        })
+        .await;
```

Comment:
This is also a very strange API; making a new `Renderer` entirely for a single render operation is likely to be extraordinarily expensive